app/api/v2/commercial/accounts.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module API
+  module V2
+    module Commercial
+      # Admin functionality over abilities
+      class Accounts < Grape::API
+        resource :accounts do
+          helpers ::API::V2::NamedParams
+          helpers do
+            def permitted_search_params(params)
+              params.slice(:keyword)
+            end
+          end
+
+          desc 'Returns array of accounts which user have access as paginated collection',
+               failure: [
+                 { code: 404, message: 'Account not found' }
+               ],
+               success: API::V2::Commercial::Entities::Account
+          params do
+            use :pagination_filters
+          end
+
+          get do
+            # Check account in the organization that user belong to
+            members = Membership.joins('LEFT JOIN organizations ON organizations.id = memberships.organization_id')
+                                .where(user_id: current_user.id)
+                                .select('memberships.*,organizations.name, organizations.parent_id')
+                                .pluck(:organization_id, :'organizations.name', :'organizations.parent_id')
+                                .map { |id, name, pid| { id: id, name: name, pid: pid } }
+
+            # Verify barong admin has AdminSwitchSession ability
+            if admin_organization? :read, AdminSwitchSession
```

Comment:
if you have admin_organization, it means only admin should have access to it
Then it should be on admin filder

Reply:
Actually, Only admin who have ability `AdminSwitchSession` and any Organization member can access this endpoint.
We use the same endpoint for both role.